stdlib/dataclasses.pyi

```diff
@@ -1,17 +1,25 @@
+import enum
 import sys
 import types
 from builtins import type as Type  # alias to avoid name clashes with fields named "type"
 from typing import Any, Callable, Generic, Iterable, Mapping, Protocol, TypeVar, overload
+from typing_extensions import Literal
 
 if sys.version_info >= (3, 9):
     from types import GenericAlias
 
 _T = TypeVar("_T")
 _T_co = TypeVar("_T_co", covariant=True)
 
-class _MISSING_TYPE: ...
+# define _MISSING_TYPE as an enum within the type stubs,
+# even though that is not really its type at runtime
+# this allows us to use Literal[MISSING]
```

Comment:
You say

> this allows us to use Literal[MISSING]

and later use:

> _T | Literal[_MISSING_TYPE.MISSING]

Which one is correct?

Reply:
Good point, `Literal[MISSING]` isn't allowed (at least unless type checkers decide something like https://github.com/python/mypy/issues/8657 is a good idea).

Both before and after this change `dataclasses._MISSING_TYPE` works as an annotation, although I expect that to be used pretty rarely.

Reply:
Nice catch, thank you! I'll amend this right away.

I missed that comment when I amended from `Literal[MISSING]` to `Literal[_MISSING_TYPE.MISSING]`.
I initially wrote the former, and was disappointed when it didn't work.